src/main/java/org/programmers/springboot/basic/domain/voucher/repository/JdbcVoucherRepository.java

```diff
@@ -0,0 +1,131 @@
+package org.programmers.springboot.basic.domain.voucher.repository;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.programmers.springboot.basic.domain.voucher.entity.Voucher;
+import org.programmers.springboot.basic.domain.voucher.entity.VoucherType;
+import org.programmers.springboot.basic.domain.voucher.service.validate.VoucherValidationStrategy;
+import org.programmers.springboot.basic.util.converter.UUIDConverter;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+@Slf4j
+@Repository
+@RequiredArgsConstructor
+public class JdbcVoucherRepository implements VoucherRepository {
+    private final JdbcTemplate jdbcTemplate;
+    private final Map<VoucherType, VoucherValidationStrategy> validationStrategyMap;
+
+    private RowMapper<Voucher> voucherRowMapper() {
+        return (rs, rowNum) -> {
+            UUID voucherId = UUIDConverter.toUUID(rs.getBytes("voucher_id"));
+            VoucherType voucherType = VoucherType.valueOf(rs.getInt("voucher_type"));
+            Long discount = rs.getLong("discount");
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+            LocalDateTime createdAt = LocalDateTime.parse(rs.getString("createdAt"), formatter);
```

Comment:
```suggestion
            LocalDateTime createdAt = LocalDateTime.parse(rs.getString("created_at"), formatter);
```

RDBMS에서 정의한 스키마의 컨벤션과 자바에서의 컨벤션이 왔다갔다 하는 것 같습니다.
DB에서는 snake_case, Java에서는 camelCase로 통일 시키는 것을 권장합니다.